pkg/clipboard/copy_darwin.go

```diff
@@ -0,0 +1,72 @@
+//go:build darwin
+// +build darwin
+
+package clipboard
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/atotto/clipboard"
+	"github.com/gopasspw/gopass/pkg/debug"
+)
+
+func copyToClipboard(ctx context.Context, content []byte) error {
+	password := string(content)
+
+	// first try to copy via osascript, if that fails fallback to pbcopy
+	if err := copyViaOsascript(ctx, password); err != nil {
+		debug.Log("failed to copy via osascript: %s", err)
+		return clipboard.WriteAll(password)
+	}
+	return nil
+}
+
+func copyViaOsascript(ctx context.Context, password string) error {
+	args := []string{
+		// The Foundation library has the Objective C symbols for pasteboard
+		"-e", `use framework "Foundation"`,
+		// Need to use scripting additions for access to "do shell script"
+		"-e", "use scripting additions",
+		// type = a reference to the ObjC constant NSPasteboardTypeString
+		// which is needed to indentify clioboard contents as text
+		"-e", "set type to current application's NSPasteboardTypeString",
+		// pb = a reference to the system's pasteboard
+		"-e", "set pb to current application's NSPasteboard's generalPasteboard()",
+		// Must clear contents before adding a new item to pasteboard
+		"-e", "pb's clearContents()",
+		// Set the flag ConcealedType so clipboard history mangers don't record the password.
+		// The first argument can by anything, but an empty string will do fine.
+		"-e", `pb's setString:"" forType:"org.nspasteboard.ConcealedType"`,
+		// AppleScript cannot read from stdin, so pipe fd#3 to stdin of cat and read the output.
+		// This output is put in the clipboard, setting type = string type
+		"-e", `pb's setString:(do shell script "cat 0<&3") forType:type`,
+	}
+	cmd := exec.CommandContext(ctx, "osascript", args...)
+	r, w, err := os.Pipe()
+	if err != nil {
+		return err
+	}
+
+	cmd.ExtraFiles = []*os.File{r} // Receiving end of pipes is connected to fd#3
+	go func() {
+		defer w.Close()
+		io.WriteString(w, password) // Write the password to fd#3
+	}()
```

Comment:
This could deserve a comment mentioning how line 60 is going to be waiting on a write on its pipe because of the above `cat 0<&3` and is therefore blocking until the `WriteString` is done, maybe. 
Also, don't we have a risk of the write "starting too soon" and the pipe not getting it all? Seems like it could be a race condition issue leading to "missing parts of password" or even it "blocking with no reason when copying" if the string is fully written before the command is actually run on line 60.

From: https://groups.google.com/g/alt.comp.lang.applescript/c/1v0s0Tkn1bM/m/b6VesyBKv6YJ
it seems we could be relying on `cat` and Stdin instead, which would allow us to get rid of the goroutine and use the "nicer":
```
cmd.Stdin = strings.NewReader(password)
```

Reply:
I couldn't get it to work with the `cmd.Stdin` approach but feel free to send a PR if you succeed.